espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/Http/Router.java

```diff
@@ -32,50 +40,64 @@ private void addRoute(Method method, String uri, RequestHandler handler) throws
         routerMap.get(method).put(uri, handler);
     }
 
-    public AppiumResponse route(IHTTPSession session) {
-        String uri = session.getUri();
-        Method method = session.getMethod();
+    public BaseResponse route(IHTTPSession session) {
+        RequestHandler handler;
+        Map<String, String> uriParams;
+
+        try {
+            String uri = session.getUri();
+            Method method = session.getMethod();
 
-        // TODO: Make this into a generic 404 Appium Response
-        RequestHandler handler = new RequestHandler() {
-            @Override
-            public AppiumResponse handle(IHTTPSession session, Map<String, String> uriParams) {
-                return null;
+            if (!routerMap.containsKey(method)) {
+                routerMap.put(method, new HashMap<String, RequestHandler>());
             }
-        };
-
-        // Get a matching handler
-        String routeUri;
-        Map<String, String> uriParams = new HashMap<String, String>();
-
-        for (Map.Entry<String, RequestHandler> entry : routerMap.get(method).entrySet()) {
-            String testUri = entry.getKey();
-            String testRegex = "^";
-            Map<Integer, String> wildcardIndices = new HashMap<Integer, String>();
-
-            int index = 0;
-            for (String uriToken : testUri.split("/")) {
-                if (uriToken.startsWith(":")) {
-                   testRegex += "/[\\w\\W]*";
-                   wildcardIndices.put(index, uriToken.substring(1));
-                } else if (!uriToken.equals("")) {
-                    testRegex += "/" + uriToken;
+
+            // By default, set handler to NotFound until we find a matching handler
+            handler = new RequestHandler() {
+                @Override
+                public BaseResponse handle(IHTTPSession session, Map<String, String> uriParams) {
+                    return new NotFoundResponse();
                 }
-                index++;
-            }
-            testRegex += "$";
+            };
+
+            // Get a matching handler
+            uriParams = new HashMap<String, String>();
 
-            if (uri.matches(testRegex)) {
-                String[] uriTokens = uri.split("/");
-                for (Map.Entry<Integer, String> wildcardIndexEntry : wildcardIndices.entrySet()) {
-                    int wildcardIndex = wildcardIndexEntry.getKey();
-                    uriParams.put(wildcardIndexEntry.getValue(), uriTokens[wildcardIndex]);
+            // Look for a matching route
+            for (Map.Entry<String, RequestHandler> entry : routerMap.get(method).entrySet()) {
```

Comment:
all of this seems kind of crazy. does nanohttpd not come with a router and a parameter extractor?